cli/azd/pkg/auth/manager.go

```diff
@@ -440,20 +440,33 @@ func (m *Manager) LoginWithDeviceCode(
 		return nil, err
 	}
 
-	m.console.MessageUxItem(ctx, &ux.MultilineMessage{
-		Lines: []string{
-			fmt.Sprintf("Start by copying the next code: %s", output.WithBold(code.UserCode())),
-			"Then press enter and continue to log in from your browser...",
-		},
-	})
-	m.console.WaitForEnter()
-
 	url := "https://microsoft.com/devicelogin"
-	if err := m.launchBrowserFn(url); err != nil {
-		log.Println("error launching browser: ", err.Error())
-		m.console.Message(ctx, fmt.Sprintf("Error launching browser. Manually go to: %s", url))
+
+	if ShouldUseCloudShellAuth() {
+		m.console.MessageUxItem(ctx, &ux.MultilineMessage{
+			Lines: []string{
+				// nolint:lll
+				"Cloud Shell is automatically authenticated under the initial account used to sign in. Run 'azd auth login' only if you need to use a different account.",
+				// nolint:lll
+				fmt.Sprintf("To sign in, use a web browser to open the page %s and enter the code %s to authenticate.", output.WithUnderline(url), output.WithBold(code.UserCode())),
```

Comment:
This doesn't match the language in the non-cloudshell auth at all, which has been altered recently. Does this need to be revisited?

I do wonder if the code could be structured differently. At the command, we can detect cloudshell auth, and print out the cloudshell disclaimer, before calling `authManager.LoginWithDeviceCode`, to have a similar login experience.

Reply:
The two flows are: 

## Local (as written today) 

1. Fetch device code
2. Show the user the device code
3. Ask the user to hit enter to try opening browser
4. If browser fails to open show the user a link 
5. Wait for completed authentication flow 

## Cloud Shell 

1. Fetch device code
2. Show user the device code and URL (cannot open browser in Cloud Shell) 
3. Wait for completed authentication flow

Since step 2 in local and step 2 in Cloud Shell are significantly different I don't see an obvious way to simplify this code.

Reply:
I'm mostly focused on the language we use in the messages displayed to the user. There's also an alternate implementation where you pass `browserFn` as a no-opt if we were interested in keeping `manager.LoginWithDeviceCode` a consistent code path.